torchtext/datasets/iwslt2016.py

```diff
@@ -29,134 +29,84 @@
-            ('en', 'de'): 1700,
-            ('en', 'fr'): 1124,
-            ('en', 'cs'): 1385,
-            ('ar', 'en'): 1702,
-            ('fr', 'en'): 1124,
-            ('de', 'en'): 1700,
-            ('cs', 'en'): 1385
+            frozenset(['ar', 'en']): 1702,
+            frozenset(['de', 'en']): 1700,
+            frozenset(['en', 'fr']): 1124,
+            frozenset(['en', 'cs']): 1385
         },
         'tst2013': {
-            ('en', 'ar'): 1169,
-            ('en', 'de'): 993,
-            ('en', 'fr'): 1026,
-            ('en', 'cs'): 1327,
-            ('ar', 'en'): 1169,
-            ('fr', 'en'): 1026,
-            ('de', 'en'): 993,
-            ('cs', 'en'): 1327
+            frozenset(['ar', 'en']): 1169,
+            frozenset(['de', 'en']): 993,
+            frozenset(['en', 'fr']): 1026,
+            frozenset(['en', 'cs']): 1327
         },
         'tst2014': {
-            ('en', 'ar'): 1107,
-            ('en', 'de'): 1305,
-            ('en', 'fr'): 1305,
-            ('ar', 'en'): 1107,
-            ('fr', 'en'): 1305,
-            ('de', 'en'): 1305
+            frozenset(['ar', 'en']): 1107,
+            frozenset(['de', 'en']): 1305,
+            frozenset(['en', 'fr']): 1305
         }
     },
     'test': {
         'dev2010': {
-            ('en', 'ar'): 887,
-            ('en', 'de'): 887,
-            ('en', 'fr'): 887,
-            ('en', 'cs'): 480,
-            ('ar', 'en'): 887,
-            ('fr', 'en'): 887,
-            ('de', 'en'): 887,
-            ('cs', 'en'): 480
+            frozenset(['ar', 'en']): 887,
+            frozenset(['de', 'en']): 887,
+            frozenset(['en', 'fr']): 887,
+            frozenset(['en', 'cs']): 480
         },
         'tst2010': {
-            ('en', 'ar'): 1569,
-            ('en', 'de'): 1565,
-            ('en', 'fr'): 1664,
-            ('en', 'cs'): 1511,
-            ('ar', 'en'): 1569,
-            ('fr', 'en'): 1664,
-            ('de', 'en'): 1565,
-            ('cs', 'en'): 1511
+            frozenset(['ar', 'en']): 1569,
+            frozenset(['de', 'en']): 1565,
+            frozenset(['en', 'fr']): 1664,
+            frozenset(['en', 'cs']): 1511
         },
         'tst2011': {
-            ('en', 'ar'): 1199,
-            ('en', 'de'): 1433,
-            ('en', 'fr'): 818,
-            ('en', 'cs'): 1013,
-            ('ar', 'en'): 1199,
-            ('fr', 'en'): 818,
-            ('de', 'en'): 1433,
-            ('cs', 'en'): 1013
+            frozenset(['ar', 'en']): 1199,
+            frozenset(['de', 'en']): 1433,
+            frozenset(['en', 'fr']): 818,
+            frozenset(['en', 'cs']): 1013
         },
         'tst2012': {
-            ('en', 'ar'): 1702,
-            ('en', 'de'): 1700,
-            ('en', 'fr'): 1124,
-            ('en', 'cs'): 1385,
-            ('ar', 'en'): 1702,
-            ('fr', 'en'): 1124,
-            ('de', 'en'): 1700,
-            ('cs', 'en'): 1385
+            frozenset(['ar', 'en']): 1702,
```

Comment:
minor point:
Keep in mind that
frozenset(['ar', 'ar', 'ar', 'en']) will key on ('ar', 'en') when doing input sanitization

If we key on tuples instead (mod sorting the input language pair) this might be avoided